src/Utilities.h

```diff
@@ -28,6 +38,40 @@ double sse_calc(const std::vector<T>& simval, const std::vector<T>& obsval){
 	
 }
 
+/**
+ * @brief Create new vector summating every Nth element
```

Comment:
There are clearly several holes in the logic of the function. It doesn't define what it should do in these cases:

1) When the length of `data_vector` is less than `n`
2) When the length of `data_vector` is not an integer multiple of `n`
3) When `n` is zero or negative

At the least, the outcome in these cases should be noted in the documentation, and ideally each should have a meaningful test case.